### PR TITLE
models

### PR DESCRIPTION
**Pull request by breakfireaus**

### CODE REVIEW

File: models/booking.js [hunk 1]
```diff
@@ -46,9 +42,8 @@ booking.init(
     timestamps: false,
     freezeTableName: true,
     underscored: true,
-    modelName: 'booking',
+    modelName: 'Booking',
```
Comment (Auralise) [inline models/booking.js]:
```suggestion
    modelName: 'booking',
```

File: models/index.js [hunk 2]
```diff
@@ -1,18 +1,58 @@
...
+  foreignKey: 'id',
+});
+//one service can have many booking
+Service.hasMany(Booking, {
+  foreignKey: 'id',
+  onDelete: 'CASCADE',
+});
+
+//many users to many industry
+users.belongsToMany(Industry, {
```
Comment (Auralise) [inline models/index.js]: Please review the name of this variable reference and please additionally review how to resolve many to many relationships as per docs: https://sequelize.org/docs/v6/advanced-association-concepts/advanced-many-to-many/

You can also look at project 13, the E-Commerce backend, for a decent example of this

File: models/industry.js [hunk 3]
```diff
@@ -26,9 +24,8 @@ industry.init(
     timestamps: false,
     freezeTableName: true,
     underscored: true,
-    modelName: 'industry',
+    modelName: 'Industry',
```
Comment (Auralise) [inline models/industry.js]:
```suggestion
    modelName: 'industry',
```

File: models/user.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
+const bcrypt = require('bcrypt');
+//todo: import bcrypt to hash the words before create and before update hook for passwords
+
+//refer to bcrypt documentation and class materials
+
+const { Model, DataTypes } = require('sequelize');
+
+const sequelize = require('../config/connection.js');
+
+class User extends Model {}
```
Comment (Auralise) [inline models/user.js]: Please add a method definition in here to check a password against the stored password using bcrypt. 

Make sure it is an async function and return await a result from the bcrypt async compare function: 
https://www.npmjs.com/package/bcrypt

File: models/user.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    },
+
+    //Description (update profile page)
+    description: {
+      type: DataTypes.TEXT('long'),
+      allowNull: false,
+    },
+
+    //Industry (update profile page)(FK)
+    industry: {
```
Comment (Auralise) [inline models/user.js]: Please remove the industry definition as the industry relationship between users is defined in UserIndustry.js

File: models/user.js [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+          );
+        }
+        return updatedUserData;
+      },
+    },
+    sequelize,
+    timestamps: false,
+    freezeTableName: true,
+    underscored: true,
+    modelName: 'User',
```
Comment (Auralise) [inline models/user.js]:
```suggestion
    modelName: 'user',
```

File: models/userIndustry.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+      primaryKey: true,
+      autoIncrement: true,
+    },
+
+    //userid (fk)
+    user_id: {
+      type: DataTypes.INTEGER,
+      //Foreign Key
+      references: {
+        model: 'User',
```
Comment (Auralise) [inline models/userIndustry.js]:
```suggestion
        model: 'user',
```

File: models/userIndustry.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+      },
+    },
+  },
+
+  {
+    sequelize,
+    timestamps: false,
+    freezeTableName: true,
+    underscored: true,
+    modelName: 'UserIndustry',
```
Comment (Auralise) [inline models/userIndustry.js]:
```suggestion
    modelName: 'user_industry',
```

File: package.json [hunk 9]
```diff
@@ -1,5 +1,6 @@
 {
   "dependencies": {
+    "@ssense/sscheduler": "^1.3.2",
```
Comment (Auralise) [inline package.json]: probably uninstall this package before the next commit/PR as we will not need it

File: models/booking.js [hunk 10]
```diff
@@ -13,21 +13,17 @@ booking.init(
...
-    },
-
-    booking_type: {
-      type: DataTypes.STRING,
-      allowNull: false,
+    //cancelled
+    cancelled: {
+      type: DataTypes.BOOLEAN,
+      defaultValue: false,
     },
```
Comment (Auralise) [inline models/booking.js:28]: Please add a confirmed field with a boolean value to track if the poster has approved the booking

File: models/booking.js [hunk 11]
```diff
@@ -13,21 +13,17 @@ booking.init(
...
-      allowNull: false,
-    },
-
-    booking_type: {
-      type: DataTypes.STRING,
-      allowNull: false,
+    //cancelled
+    cancelled: {
+      type: DataTypes.BOOLEAN,
+      defaultValue: false,
```
Comment (Auralise) [inline models/booking.js:26]: Please make this field not nullable

File: models/service.js [hunk 12]
```diff
@@ -20,57 +14,64 @@ service.init(
...
       allowNull: false,
     },
-    //availability
-    availability: {
-        type: DataTypes.STRING,
-        allowNull: false,
+
+    //service start
+    service_start: {
+      type: DataTypes.STRING,
```
Comment (Auralise) [inline models/service.js]: Please make this a datetime datatype (DataType.DATE: https://sequelize.org/api/v6/variable/index.html#static-variable-DataTypes)

Comment (Auralise) [inline models/service.js]:
```suggestion
      type: DataTypes.DATE,
```

File: models/service.js [hunk 13]
```diff
@@ -20,57 +14,64 @@ service.init(
...
+      allowNull: false,
+    },
+
+    //service end to do
+    service_end: {
+      type: DataTypes.STRING,
```
Comment (Auralise) [inline models/service.js]:
```suggestion
      type: DataTypes.DATE,
```

File: models/service.js [hunk 14]
```diff
@@ -20,57 +14,64 @@ service.init(
...
         type: DataTypes.DECIMAL,
         allowNull: false,
-        },
+      },
 
-    //short description(list service page)
-    service_desc: {
+      //short description(list service page)
+      service_desc: {
         type: DataTypes.INTEGER,
```
Comment (Auralise) [inline models/service.js:70]:
```suggestion
        type: DataTypes.TEXT,
```

File: models/service.js [hunk 15]
```diff
@@ -20,57 +14,64 @@ service.init(
...
+    },
   },
 
   {
     sequelize,
     timestamps: false,
     freezeTableName: true,
     underscored: true,
-    modelName: 'service',
+    modelName: 'Service',
```
Comment (Auralise) [inline models/service.js]:
```suggestion
    modelName: 'service',
```

File: models/user.js [hunk 16]
```diff
@@ -0,0 +1,88 @@
...
+      allowNull: false,
+      validate: {
+        isEmail: true,
+      },
+    },
+
+    //Password
+    password: {
+      type: DataTypes.STRING,
+      allowNull: false,
```
Comment (Auralise) [inline models/user.js:50]: Please add either a length validator (either len or min) to the password: https://sequelize.org/docs/v6/core-concepts/validations-and-constraints/#validators

File: models/service.js (hunk 12 above)
Comment (Auralise) [reply models/service.js]: Please also add some validation to these to make sure they are dates: https://sequelize.org/docs/v6/core-concepts/validations-and-constraints/#validators

**Review by Auralise — CHANGES_REQUESTED** [review]
Generally looking good. I have added a few direct comments to the files to help guide where I feel changes should be made. 

I have directly suggested that all of the sequelize model names lowercase (so that they match) as these were mismatched in a few places which would have caused us issues. This is different to the JS class names which are capitalised by convention 

I have added specific comments for changes and their reason to the files/lines directly. For simple changes, I have provided specific changes to make but for the more complex items, I have provided links to the documentation. 

Good work, keep it up!

File: models/booking.js (hunk 11 above)
Comment (breakfireaus) [reply models/booking.js:26]: done

File: models/booking.js (hunk 10 above)
Comment (breakfireaus) [reply models/booking.js:28]: done

File: models/service.js (hunk 12 above)
Comment (breakfireaus) [reply models/service.js]: done

File: models/user.js (hunk 4 above)
Comment (breakfireaus) [reply models/user.js]: it is already in the hook, spoke to Chee about it just got to add a validator

File: models/user.js (hunk 16 above)
Comment (breakfireaus) [reply models/user.js:50]: done

File: models/user.js (hunk 5 above)
Comment (breakfireaus) [reply models/user.js]: done

**Review by Auralise — APPROVED** [review]
Happy with the changes, merging in now

**Pull request merged**